### DISCUSSION TITLE
ModemManager service

### DISCUSSION THREAD

**Discussion by root2185** [davmac314/dinit#224] [Q&A] [answered]
I see that there are runit and s6 service packages, in Artix repo, but no dinit service.
I made one and it didn't work, maybe because it is already running.
Something starts it and restarts it if I kill it.

```
ps -feww | grep "ModemMa"
root     13104 13089  0 19:36 ?        00:00:00 dbus-monitor --system interface='org.freedesktop.ModemManager1.Modem.Voice',type='signal',member='CallAdded'
root     13107 13089  0 19:36 ?        00:00:00 dbus-monitor --system interface='org.freedesktop.ModemManager1.Modem.Messaging',type='signal',member='Added'
root     13111 13089  0 19:36 ?        00:00:00 dbus-monitor --system interface='org.freedesktop.DBus.Properties',member='PropertiesChanged',arg0='org.freedesktop.ModemManager1.Call'
root     13117 13089  0 19:36 ?        00:00:00 dbus-monitor --system interface='org.freedesktop.ModemManager1.Modem',type='signal',member='StateChanged'
root     22109 22080 99 20:32 pts/0    00:00:00 grep ModemMa
root     23581     1  0 15:29 ?        00:00:14 /usr/bin/ModemManager
```

**Comment by mobin-2008** [comment]
Hi. First of all I think artix forum is better place to post your issue (forum.artixlinux.org).
I suspect that ModemManager got launched by your DE (like GNOME) or something else.

**Reply by root2185** [reply]
I use a standalone WM.
Even when there is only a tty, it is restarted.
 I saw `starting in system bus` after pkilling it.
Is that dbus?

Can I control it using dinit?
For example I want to shut it down sometimes, but it is always restarted.

**Reply by davmac314** [reply]
Yep, sounds like it is being started by DBus. The S6/runit services are most likely allowing to start it early, but that's probably unnecessary anyway.

**Reply by root2185** [reply]
Do you know how can I prevent that?
I'd like to use dinit instead.

**Reply by davmac314** [reply]
https://dbus.freedesktop.org/doc/dbus-daemon.1.html

See "INTEGRATING SYSTEM SERVICES"

**Reply by root2185** [reply, marked as answer]
I deleted `/usr/share/dbus-1/system-services/org.freedesktop.ModemManager1.service` and now the dinit service works, thanks.

**Comment by root2185** [comment]
```
type            = process
command         = /usr/bin/ModemManager --debug 
logfile         = /var/log/dinit/modemmanager.log
restart         = true
waits-for       = loginready
```
I see output in the tty.
Can I redirect it all to the log?

**Comment by mobin-2008** [comment]
Hmm, from ModemManager man-page (https://www.freedesktop.org/software/ModemManager/man/latest/ModemManager.8.html):

> --debug Runs ModemManager with "DEBUG" log level and without daemonizing. This is useful for debugging, as it directs log output to the controlling terminal in addition to syslog.


I'm not sure it is related or not.

**Reply by root2185** [reply]
Right, with:
```
command         = /usr/bin/ModemManager --debug  --log-file=/var/log/dinit/modemmanager.log
```
It does not clutter the tty anymore.

I am worried about the log getting million of lines though.
Is it cleaned automatically after some time?

**Reply by mobin-2008** [reply]
> Right, with:
> 
> ```
> command         = /usr/bin/ModemManager --debug  --log-file=/var/log/dinit/modemmanager.log
> ```
> 
> It does not clutter the tty anymore.
> 
> I am worried about the log getting million of lines though. Is it cleaned automatically after some time?


No, It's not I guess. but if you don't care about debug logs you can just send its log into /dev/null (--log-file=/dev/null) or use [`logrotate`]( https://linux.die.net/man/8/logrotate).
